ui/client/components/graph/node-modal/editors/expression/EditorType.js

```diff
@@ -1,41 +1,47 @@
-import {parseableBoolean} from "./ExpressionParser"
 import RawEditor from "./RawEditor"
 import BoolEditor from "./BoolEditor"
+import StringEditor from "./StringEditor"
 import ExpressionWithFixedValues from "./ExpressionWithFixedValues"
 
 class EditorType {
 
-  isSupported = (filedType) => supportedFieldType.includes(filedType)
+  isSupported = (fieldType) => Object.values(editors).map(value => value.supportedFieldType).includes(fieldType)
 
-  editorName = (fieldType, expressionObj, displayRawEditor) => {
-    switch (fieldType) {
-      case Types.BOOLEAN:
-        return !displayRawEditor && this.switchableToBooleanEditor(expressionObj) ? Types.BOOL_EDITOR : Types.RAW_EDITOR
-      case Types.EXPRESSION_WITH_FIXED_VALUES || Types.RAW_EDITOR:
-        return fieldType
-      default:
-        return Types.RAW_EDITOR
-    }
-  }
-
-  switchableToBooleanEditor = (expressionObj) => parseableBoolean(expressionObj) || _.isEmpty(expressionObj.expression)
-
-  editor = (editorName) => _.get(editorTypes, editorName, editorTypes[Types.RAW_EDITOR])
+  editor = (editorName) => _.get(editors, editorName, editors[Types.RAW_EDITOR])
 }
 
 export const Types = {
   BOOL_EDITOR: "boolEditor",
+  STRING_EDITOR: "stringEditor",
   RAW_EDITOR: "rawEditor",
+  EXPRESSION: "expression",
   BOOLEAN: "Boolean",
+  STRING: "String",
   EXPRESSION_WITH_FIXED_VALUES: "expressionWithFixedValues",
 }
 
-const supportedFieldType = [Types.BOOLEAN]
-
-const editorTypes = {
-  [Types.RAW_EDITOR]: RawEditor,
-  [Types.BOOL_EDITOR]: BoolEditor,
-  [Types.EXPRESSION_WITH_FIXED_VALUES]: ExpressionWithFixedValues
+const editors = {
+  [Types.RAW_EDITOR]: {
+    editor: RawEditor,
+    switchable: (editorName, expressionObj, fieldType) => RawEditor.switchableFrom(editorName, expressionObj, fieldType),
+    switchableHint: (editorName, expressionObj, fieldType) => RawEditor.switchableFromHint(editorName, expressionObj, fieldType),
+    supportedFieldType: Types.BOOLEAN
+  },
+  [Types.BOOL_EDITOR]: {
+    editor: BoolEditor,
+    switchable: BoolEditor.switchableFrom(),
+    switchableHint: (_) => BoolEditor.switchableFromHint,
+  },
+  [Types.STRING_EDITOR]: {
+    editor: StringEditor,
+    switchable: BoolEditor.switchableFrom(),
+    switchableHint: (_) => StringEditor.switchableFromHint,
```

Comment:
but it should be other way round:
- switchableTo 
- switchableToHint
- notSwitchableToHint

Reply:
Done